### PR TITLE
[full-ci] standalone graph service with LDAP

### PR DESCRIPTION
**Pull request by butonic**
This PR allows setting an API Token for bearer based authentication. Setting a Token will disable the requireAdmin middleware and  for now, also disable provisioning roles for new users.

Together with the LDAP backend this allows using the graph service for provisioning.

we will fix creation status codes in a subsequent PR

### CODE REVIEW

**Comment by butonic** [comment]
Notes:
- `currentUser := ctxpkg.ContextMustGetUser(r.Context())` is used in several places. If we cannot get the user we should check an access token was used (and debug log and trace the request)
- the requests that require an authenticated user in the context won't work, eg the /me endpoints ... maybe a different router is easier than trying to disable certain endpoints.

**Comment by ownclouders** [comment]
:boom: Acceptance test [<strong>localApiTests-apiArchiver-ocis</strong>](https://drone.owncloud.com/owncloud/ocis/17355/46/1) failed. Further test are cancelled...

**Comment by butonic** [comment]
wow, CI ran into a panic https://drone.owncloud.com/owncloud/ocis/17385/51/4
```
github.com/cs3org/reva/v2/pkg/storage/utils/decomposedfs/node.ReadNode({0xc00087ed20?, 0x400?}, {0x3adc310?, 0xc000f32288}, {0xc0026e0270, 0x24}, {0xc0026e0240, 0x24}, 0x0)
	/go/pkg/mod/github.com/cs3org/reva/v2@v2.12.0/pkg/storage/utils/decomposedfs/node/node.go:311 +0x777
github.com/cs3org/reva/v2/pkg/storage/utils/decomposedfs/lookup.(*Lookup).NodeFromID(0x203000?, {0x3ae2f98?, 0xc0024922a0?}, 0x3aed528?)
	/go/pkg/mod/github.com/cs3org/reva/v2@v2.12.0/pkg/storage/utils/decomposedfs/lookup/lookup.go:77 +0x65
github.com/cs3org/reva/v2/pkg/storage/utils/decomposedfs/lookup.(*Lookup).NodeFromResource(0xc002d9c700?, {0x3ae2f98, 0xc0024922a0}, 0xc002d9c480)
	/go/pkg/mod/github.com/cs3org/reva/v2@v2.12.0/pkg/storage/utils/decomposedfs/lookup/lookup.go:45 +0x53
github.com/cs3org/reva/v2/pkg/storage/utils/decomposedfs.(*Decomposedfs).GetMD(0xc000733810, {0x3ae2f98, 0xc0024922a0}, 0xc002d9c480, {0x0, 0x0, 0x0}, {0x0, 0x0, 0x0})
	/go/pkg/mod/github.com/cs3org/reva/v2@v2.12.0/pkg/storage/utils/decomposedfs/decomposedfs.go:481 +0x69
github.com/cs3org/reva/v2/internal/grpc/services/storageprovider.(*service).Stat(0xc0008faf00, {0x3ae2f98, 0xc002492270}, 0xc0012c0dc0)
	/go/pkg/mod/github.com/cs3org/reva/v2@v2.12.0/internal/grpc/services/storageprovider/storageprovider.go:691 +0x29f
github.com/cs3org/go-cs3apis/cs3/storage/provider/v1beta1._ProviderAPI_Stat_Handler.func1({0x3ae2f98, 0xc002492270}, {0x27871a0?, 0xc0012c0dc0})
	/go/pkg/mod/github.com/cs3org/go-cs3apis@v0.0.0-20221012090518-ef2996678965/cs3/storage/provider/v1beta1/provider_api.pb.go:6238 +0x7b
go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc.UnaryServerInterceptor.func1({0x3ae2f98, 0xc002492150}, {0x27871a0, 0xc0012c0dc0}, 0xc0018a9860, 0xc0065ecaf8)
	/go/pkg/mod/go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc@v0.36.4/interceptor.go:341 +0x437
github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1({0x3ae2f98?, 0xc002492150?}, {0x27871a0?, 0xc0012c0dc0?})
	/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:25 +0x3a
github.com/cs3org/reva/v2/internal/grpc/interceptors/eventsmiddleware.NewUnary.func1({0x3ae2f98, 0xc002492120}, {0x27871a0?, 0xc0012c0dc0?}, 0x0?, 0xc0018a9880)
	/go/pkg/mod/github.com/cs3org/reva/v2@v2.12.0/internal/grpc/interceptors/eventsmiddleware/events.go:72 +0xe8
github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1({0x3ae2f98?, 0xc002492120?}, {0x27871a0?, 0xc0012c0dc0?})
	/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:25 +0x3a
github.com/cs3org/reva/v2/internal/grpc/interceptors/prometheus.interceptorFromConfig.func1({0x3ae2f98?, 0xc002492120?}, {0x27871a0?, 0xc0012c0dc0?}, 0x23a9b40?, 0x3abcd01?)
	/go/pkg/mod/github.com/cs3org/reva/v2@v2.12.0/internal/grpc/interceptors/prometheus/prometheus.go:72 +0x43
github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1({0x3ae2f98?, 0xc002492120?}, {0x27871a0?, 0xc0012c0dc0?})
	/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:25 +0x3a
github.com/cs3org/reva/v2/internal/grpc/interceptors/auth.NewUnary.func2({0x3ae2f98, 0xc002625e30}, {0x27871a0, 0xc0012c0dc0}, 0xc0018a9860, 0xc0018a98e0)
	/go/pkg/mod/github.com/cs3org/reva/v2@v2.12.0/internal/grpc/interceptors/auth/auth.go:160 +0xa06
github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1({0x3ae2f98?, 0xc002625e30?}, {0x27871a0?, 0xc0012c0dc0?})
	/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:25 +0x3a
github.com/grpc-ecosystem/go-grpc-middleware/recovery.UnaryServerInterceptor.func1({0x3ae2f98?, 0xc002625e30?}, {0x27871a0?, 0xc0012c0dc0?}, 0xc002223208?, 0x496757?)
	/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/recovery/interceptors.go:33 +0xc4
github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1({0x3ae2f98?, 0xc002625e30?}, {0x27871a0?, 0xc0012c0dc0?})
	/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:25 +0x3a
github.com/cs3org/reva/v2/internal/grpc/interceptors/log.NewUnary.func1({0x3ae2f98, 0xc002625e30}, {0x27871a0, 0xc0012c0dc0}, 0xc0018a9860, 0xc0018a9920)
	/go/pkg/mod/github.com/cs3org/reva/v2@v2.12.0/internal/grpc/interceptors/log/log.go:39 +0x9a
github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1({0x3ae2f98?, 0xc002625e30?}, {0x27871a0?, 0xc0012c0dc0?})
	/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:25 +0x3a
github.com/cs3org/reva/v2/internal/grpc/interceptors/useragent.NewUnary.func1({0x3ae2f98, 0xc002625d70}, {0x27871a0, 0xc0012c0dc0}, 0x1?, 0xc0018a9940)
	/go/pkg/mod/github.com/cs3org/reva/v2@v2.12.0/internal/grpc/interceptors/useragent/useragent.go:38 +0xe8
github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1({0x3ae2f98?, 0xc002625d70?}, {0x27871a0?, 0xc0012c0dc0?})
	/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:25 +0x3a
github.com/cs3org/reva/v2/internal/grpc/interceptors/token.NewUnary.func1({0x3ae2f98, 0xc002625c50}, {0x27871a0, 0xc0012c0dc0}, 0x5196aa8?, 0xc0018a9960)
	/go/pkg/mod/github.com/cs3org/reva/v2@v2.12.0/internal/grpc/interceptors/token/token.go:44 +0x158
github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1({0x3ae2f98?, 0xc002625c50?}, {0x27871a0?, 0xc0012c0dc0?})
	/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:25 +0x3a
github.com/cs3org/reva/v2/internal/grpc/interceptors/appctx.NewUnary.func1({0x3ae2f98, 0xc002625b60}, {0x27871a0, 0xc0012c0dc0}, 0x7f699474dd08?, 0xc0018a9980)
	/go/pkg/mod/github.com/cs3org/reva/v2@v2.12.0/internal/grpc/interceptors/appctx/appctx.go:51 +0x73c
github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1({0x3ae2f98?, 0xc002625b60?}, {0x27871a0?, 0xc0012c0dc0?})
	/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:25 +0x3a
github.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1({0x3ae2f98, 0xc002625b60}, {0x27871a0, 0xc0012c0dc0}, 0xc002320a18?, 0x24a66e0?)
	/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:34 +0xbf
github.com/cs3org/go-cs3apis/cs3/storage/provider/v1beta1._ProviderAPI_Stat_Handler({0x2857ba0?, 0xc0008faf00}, {0x3ae2f98, 0xc002625b60}, 0xc002319340, 0xc00129a6c0)
	/go/pkg/mod/github.com/cs3org/go-cs3apis@v0.0.0-20221012090518-ef2996678965/cs3/storage/provider/v1beta1/provider_api.pb.go:6240 +0x138
google.golang.org/grpc.(*Server).processUnaryRPC(0xc001d541e0, {0x3af2990, 0xc001b96820}, 0xc0023a39e0, 0xc00129a870, 0x4fb8df0, 0x0)
	/go/pkg/mod/google.golang.org/grpc@v1.51.0/server.go:1340 +0xd13
google.golang.org/grpc.(*Server).handleStream(0xc001d541e0, {0x3af2990, 0xc001b96820}, 0xc0023a39e0, 0x0)
	/go/pkg/mod/google.golang.org/grpc@v1.51.0/server.go:1713 +0xa1b
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	/go/pkg/mod/google.golang.org/grpc@v1.51.0/server.go:965 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	/go/pkg/mod/google.golang.org/grpc@v1.51.0/server.go:963 +0x28a
{"level":"error","service":"storage-users","pkg":"rgrpc","traceid":"00000000000000000000000000000000","time":"2022-12-08T13:20:03.260346195Z","message":"runtime error: invalid memory address or nil pointer dereference; stack: goroutine 104980 [running]:\nruntime/debug.Stack()\n\t/usr/local/go/src/runtime/debug/stack.go:24 +0x65\ngithub.com/cs3org/reva/v2/internal/grpc/interceptors/recovery.recoveryFunc({0x3ae2f98, 0xc002625e30}, {0x24a7a60, 0x4f971b0})\n\t/go/pkg/mod/github.com/cs3org/reva/v2@v2.12.0/internal/grpc/interceptors/recovery/recovery.go:50 +0x65\ngithub.com/grpc-ecosystem/go-grpc-middleware/recovery.recoverFrom({0x3ae2f98?, 0xc002625e30?}, {0x24a7a60?, 0x4f971b0?}, 0xc000cc1b50?)\n\t/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/recovery/interceptors.go:61 +0x36\ngithub.com/grpc-ecosystem/go-grpc-middleware/recovery.UnaryServerInterceptor.func1.1()\n\t/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/recovery/interceptors.go:29 +0x7b\npanic({0x24a7a60, 0x4f971b0})\n\t/usr/local/go/src/runtime/panic.go:838 +0x207\ngithub.com/cs3org/reva/v2/pkg/storage/utils/decomposedfs/node.ReadNode.func1()\n\t/go/pkg/mod/github.com/cs3org/reva/v2@v2.12.0/pkg/storage/utils/decomposedfs/node/node.go:241 +0x28\ngithub.com/cs3org/reva/v2/pkg/storage/utils/decomposedfs/node.ReadNode({0xc00087ed20?, 0x400?}, {0x3adc310?, 0xc000f32288}, {0xc0026e0270, 0x24}, {0xc0026e0240, 0x24}, 0x0)\n\t/go/pkg/mod/github.com/cs3org/reva/v2@v2.12.0/pkg/storage/utils/decomposedfs/node/node.go:311 +0x777\ngithub.com/cs3org/reva/v2/pkg/storage/utils/decomposedfs/lookup.(*Lookup).NodeFromID(0x203000?, {0x3ae2f98?, 0xc0024922a0?}, 0x3aed528?)\n\t/go/pkg/mod/github.com/cs3org/reva/v2@v2.12.0/pkg/storage/utils/decomposedfs/lookup/lookup.go:77 +0x65\ngithub.com/cs3org/reva/v2/pkg/storage/utils/decomposedfs/lookup.(*Lookup).NodeFromResource(0xc002d9c700?, {0x3ae2f98, 0xc0024922a0}, 0xc002d9c480)\n\t/go/pkg/mod/github.com/cs3org/reva/v2@v2.12.0/pkg/storage/utils/decomposedfs/lookup/lookup.go:45 +0x53\ngithub.com/cs3org/reva/v2/pkg/storage/utils/decomposedfs.(*Decomposedfs).GetMD(0xc000733810, {0x3ae2f98, 0xc0024922a0}, 0xc002d9c480, {0x0, 0x0, 0x0}, {0x0, 0x0, 0x0})\n\t/go/pkg/mod/github.com/cs3org/reva/v2@v2.12.0/pkg/storage/utils/decomposedfs/decomposedfs.go:481 +0x69\ngithub.com/cs3org/reva/v2/internal/grpc/services/storageprovider.(*service).Stat(0xc0008faf00, {0x3ae2f98, 0xc002492270}, 0xc0012c0dc0)\n\t/go/pkg/mod/github.com/cs3org/reva/v2@v2.12.0/internal/grpc/services/storageprovider/storageprovider.go:691 +0x29f\ngithub.com/cs3org/go-cs3apis/cs3/storage/provider/v1beta1._ProviderAPI_Stat_Handler.func1({0x3ae2f98, 0xc002492270}, {0x27871a0?, 0xc0012c0dc0})\n\t/go/pkg/mod/github.com/cs3org/go-cs3apis@v0.0.0-20221012090518-ef2996678965/cs3/storage/provider/v1beta1/provider_api.pb.go:6238 +0x7b\ngo.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc.UnaryServerInterceptor.func1({0x3ae2f98, 0xc002492150}, {0x27871a0, 0xc0012c0dc0}, 0xc0018a9860, 0xc0065ecaf8)\n\t/go/pkg/mod/go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc@v0.36.4/interceptor.go:341 +0x437\ngithub.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1({0x3ae2f98?, 0xc002492150?}, {0x27871a0?, 0xc0012c0dc0?})\n\t/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:25 +0x3a\ngithub.com/cs3org/reva/v2/internal/grpc/interceptors/eventsmiddleware.NewUnary.func1({0x3ae2f98, 0xc002492120}, {0x27871a0?, 0xc0012c0dc0?}, 0x0?, 0xc0018a9880)\n\t/go/pkg/mod/github.com/cs3org/reva/v2@v2.12.0/internal/grpc/interceptors/eventsmiddleware/events.go:72 +0xe8\ngithub.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1({0x3ae2f98?, 0xc002492120?}, {0x27871a0?, 0xc0012c0dc0?})\n\t/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:25 +0x3a\ngithub.com/cs3org/reva/v2/internal/grpc/interceptors/prometheus.interceptorFromConfig.func1({0x3ae2f98?, 0xc002492120?}, {0x27871a0?, 0xc0012c0dc0?}, 0x23a9b40?, 0x3abcd01?)\n\t/go/pkg/mod/github.com/cs3org/reva/v2@v2.12.0/internal/grpc/interceptors/prometheus/prometheus.go:72 +0x43\ngithub.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1({0x3ae2f98?, 0xc002492120?}, {0x27871a0?, 0xc0012c0dc0?})\n\t/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:25 +0x3a\ngithub.com/cs3org/reva/v2/internal/grpc/interceptors/auth.NewUnary.func2({0x3ae2f98, 0xc002625e30}, {0x27871a0, 0xc0012c0dc0}, 0xc0018a9860, 0xc0018a98e0)\n\t/go/pkg/mod/github.com/cs3org/reva/v2@v2.12.0/internal/grpc/interceptors/auth/auth.go:160 +0xa06\ngithub.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1({0x3ae2f98?, 0xc002625e30?}, {0x27871a0?, 0xc0012c0dc0?})\n\t/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:25 +0x3a\ngithub.com/grpc-ecosystem/go-grpc-middleware/recovery.UnaryServerInterceptor.func1({0x3ae2f98?, 0xc002625e30?}, {0x27871a0?, 0xc0012c0dc0?}, 0xc002223208?, 0x496757?)\n\t/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/recovery/interceptors.go:33 +0xc4\ngithub.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1({0x3ae2f98?, 0xc002625e30?}, {0x27871a0?, 0xc0012c0dc0?})\n\t/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:25 +0x3a\ngithub.com/cs3org/reva/v2/internal/grpc/interceptors/log.NewUnary.func1({0x3ae2f98, 0xc002625e30}, {0x27871a0, 0xc0012c0dc0}, 0xc0018a9860, 0xc0018a9920)\n\t/go/pkg/mod/github.com/cs3org/reva/v2@v2.12.0/internal/grpc/interceptors/log/log.go:39 +0x9a\ngithub.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1({0x3ae2f98?, 0xc002625e30?}, {0x27871a0?, 0xc0012c0dc0?})\n\t/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:25 +0x3a\ngithub.com/cs3org/reva/v2/internal/grpc/interceptors/useragent.NewUnary.func1({0x3ae2f98, 0xc002625d70}, {0x27871a0, 0xc0012c0dc0}, 0x1?, 0xc0018a9940)\n\t/go/pkg/mod/github.com/cs3org/reva/v2@v2.12.0/internal/grpc/interceptors/useragent/useragent.go:38 +0xe8\ngithub.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1({0x3ae2f98?, 0xc002625d70?}, {0x27871a0?, 0xc0012c0dc0?})\n\t/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:25 +0x3a\ngithub.com/cs3org/reva/v2/internal/grpc/interceptors/token.NewUnary.func1({0x3ae2f98, 0xc002625c50}, {0x27871a0, 0xc0012c0dc0}, 0x5196aa8?, 0xc0018a9960)\n\t/go/pkg/mod/github.com/cs3org/reva/v2@v2.12.0/internal/grpc/interceptors/token/token.go:44 +0x158\ngithub.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1({0x3ae2f98?, 0xc002625c50?}, {0x27871a0?, 0xc0012c0dc0?})\n\t/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:25 +0x3a\ngithub.com/cs3org/reva/v2/internal/grpc/interceptors/appctx.NewUnary.func1({0x3ae2f98, 0xc002625b60}, {0x27871a0, 0xc0012c0dc0}, 0x7f699474dd08?, 0xc0018a9980)\n\t/go/pkg/mod/github.com/cs3org/reva/v2@v2.12.0/internal/grpc/interceptors/appctx/appctx.go:51 +0x73c\ngithub.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1.1.1({0x3ae2f98?, 0xc002625b60?}, {0x27871a0?, 0xc0012c0dc0?})\n\t/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:25 +0x3a\ngithub.com/grpc-ecosystem/go-grpc-middleware.ChainUnaryServer.func1({0x3ae2f98, 0xc002625b60}, {0x27871a0, 0xc0012c0dc0}, 0xc002320a18?, 0x24a66e0?)\n\t/go/pkg/mod/github.com/grpc-ecosystem/go-grpc-middleware@v1.3.0/chain.go:34 +0xbf\ngithub.com/cs3org/go-cs3apis/cs3/storage/provider/v1beta1._ProviderAPI_Stat_Handler({0x2857ba0?, 0xc0008faf00}, {0x3ae2f98, 0xc002625b60}, 0xc002319340, 0xc00129a6c0)\n\t/go/pkg/mod/github.com/cs3org/go-cs3apis@v0.0.0-20221012090518-ef2996678965/cs3/storage/provider/v1beta1/provider_api.pb.go:6240 +0x138\ngoogle.golang.org/grpc.(*Server).processUnaryRPC(0xc001d541e0, {0x3af2990, 0xc001b96820}, 0xc0023a39e0, 0xc00129a870, 0x4fb8df0, 0x0)\n\t/go/pkg/mod/google.golang.org/grpc@v1.51.0/server.go:1340 +0xd13\ngoogle.golang.org/grpc.(*Server).handleStream(0xc001d541e0, {0x3af2990, 0xc001b96820}, 0xc0023a39e0, 0x0)\n\t/go/pkg/mod/google.golang.org/grpc@v1.51.0/server.go:1713 +0xa1b\ngoogle.golang.org/grpc.(*Server).serveStreams.func1.2()\n\t/go/pkg/mod/google.golang.org/grpc@v1.51.0/server.go:965 +0x98\ncreated by google.golang.org/grpc.(*Server).serveStreams.func1\n\t/go/pkg/mod/google.golang.org/grpc@v1.51.0/server.go:963 +0x28a\n"}
{"level":"error","service":"storage-users","pkg":"rgrpc","traceid":"00000000000000000000000000000000","user-agent":"GuzzleHttp/7","from":"tcp://127.0.0.1:38868","uri":"/cs3.storage.provider.v1beta1.ProviderAPI/Stat","start":"08/Dec/2022:13:20:03 +0000","end":"08/Dec/2022:13:20:03 +0000","time_ns":999286,"code":"Internal","time":"2022-12-08T13:20:03.26040212Z","message":"unary"}
```

**Comment by butonic** [comment]
fixed the panic in https://github.com/cs3org/reva/pull/3534

**Comment by rhafer** [comment]
@butonic I was able to reproduce the webui test failure. And AFAICS it's caused by the changed status code in PostUser(). I have no idea why that is. But changing the status code back to `ok` makes the test  succeed again :man_shrugging: 

I somehow suspect this to be a weird issue with test-middleware ... but that is just a wild guess.

**Comment by butonic** [comment]
aaaargh: https://github.com/owncloud/owncloud-test-middleware/blob/a8233f8f5895a74d09e83b75199883c62a8bbf8a/src/helpers/graphHelper.js#L28

**Comment by butonic** [comment]
thx, we should get this part of the changes in ASAP

**Comment by butonic** [comment]
to test set up an ldap server and set these envirenment variables:
```

					// disable events
					"GRAPH_EVENTS_ENDPOINT": "",

					// directly talk to ldap
					"GRAPH_IDENTITY_BACKEND": "ldap", // ldap is the default
					
					// LDAP connection
					"GRAPH_LDAP_URI": "ldap://ldap",
					"GRAPH_LDAP_CACERT": "",
					"GRAPH_LDAP_INSECURE": "true",
					"GRAPH_LDAP_BIND_DN": "cn=admin,dc=owncloud,dc=com",
					"GRAPH_LDAP_BIND_PASSWORD": "admin",

					"GRAPH_LDAP_SERVER_UUID": "true",
					"GRAPH_LDAP_SERVER_USE_PASSWORD_MODIFY_EXOP": "false",
					"GRAPH_LDAP_SERVER_WRITE_ENABLED": "true",

					// schema definition
					"GRAPH_LDAP_USER_BASE_DN": "ou=users,dc=owncloud,dc=com",
					"GRAPH_LDAP_USER_SCOPE": "sub",
					"GRAPH_LDAP_USER_FILTER": "",
					"GRAPH_LDAP_USER_OBJECTCLASS": "inetOrgPerson",
					"GRAPH_LDAP_USER_EMAIL_ATTRIBUTE": "mail",
					"GRAPH_LDAP_USER_DISPLAYNAME_ATTRIBUTE": "displayName",
					"GRAPH_LDAP_USER_NAME_ATTRIBUTE": "uid",
					"GRAPH_LDAP_USER_UID_ATTRIBUTE": "entryUUID",
					"GRAPH_LDAP_GROUP_BASE_DN": "ou=groups,dc=owncloud,dc=com",
					"GRAPH_LDAP_GROUP_SEARCH_SCOPE": "sub",
					"GRAPH_LDAP_GROUP_FILTER": "",
					"GRAPH_LDAP_GROUP_OBJECTCLASS": "groupOfNames",
					"GRAPH_LDAP_GROUP_NAME_ATTRIBUTE": "cn",
					"GRAPH_LDAP_GROUP_ID_ATTRIBUTE": "entryUUID",

					// we use an api token to authenticate requests
					"GRAPH_HTTP_API_TOKEN": "a-secret-api-token",

					// the jwt is not needed for the ldap backend
					"GRAPH_JWT_SECRET":"unused",
```

**Review by micbar — APPROVED** [review]
LGTM

File: services/graph/pkg/server/http/server.go [hunk 1]
```diff
@@ -82,25 +86,49 @@ func Server(opts ...Option) (http.Service, error) {
...
 				account.JWTSecret(options.Config.TokenManager.JWTSecret),
-			),
-		),
+			))
+		roleService = settingssvc.NewRoleService("com.owncloud.api.settings", grpc.DefaultClient())
+		gatewayClient, err = pool.GetGatewayServiceClient(options.Config.Reva.Address, options.Config.Reva.GetRevaOptions()...)
+		if err != nil {
+			return http.Service{}, errors.Wrap(err, "could not initialize gateway client")
+		}
+	} else {
```
Comment (C0rby) [inline services/graph/pkg/server/http/server.go:115]: Is the `requireAdminMiddleware` missing here?

Comment (butonic) [reply services/graph/pkg/server/http/server.go:115]: no, when it is nil the handler will initialize a requireAdmin middleware itself ... we could move that into the server.go, but then we should do that in a consistent way for all middlewares. Well, maybe one at a time.

In any case, this works as intended.

**Pull request merged**